src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,70 @@
+package duke.storage;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import duke.exceptions.DukeIoException;
+/**
+ * The Storage class handles file storage operations for the Duke chatbot application.
+ */
+public class Storage {
+
+    private final String path;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param path The path to the file.
+     * @throws DukeIoException If an error occurs during file creation.
+     */
+    public Storage(String path) {
```

Comment:
Perhaps your Constructor could explicitly declare using the `throws` clause in the method signature?
```suggestion
    public Storage(String path) throws DukeIoException {
```